backend/google_search.py

```diff
@@ -1,53 +1,70 @@
+from typing import List
+import os
+
+from dotenv import load_dotenv
 import requests
 
+from helper_classes.CustomSearchResult import CustomSearchResult
+
+
+# load env files
+load_dotenv()
+
 # API credentials
-api_key = "AIzaSyA_V85LyvaxhPceZneJ3tzrKRZklqcI-WA"
-cx = "a10036247dff048e4"
-num_results = 20
+API_URL = os.getenv("CUSTOM_SEARCH_API_URL")
+API_KEY = os.getenv("CUSTOM_SEARCH_API_KEY")
+CX = os.getenv("CUSTOM_SEARCH_CX")
+
 
-def search(query_list):
+def custom_search(query_list: List[str]) -> List[CustomSearchResult]:
```

Comment:
Ikkada query list enduku pass cheyyadam? We can just pass a single query, it will search for a single query. Kaavali ante, different queries tho function ni malli malli call cheyyochu.

Reply:
Tharvatha next commit lo change cheddham